### DISCUSSION TITLE
flipDMABufferIfReady();

### DISCUSSION THREAD

**Discussion by Westworld** [LAutour/ESP32-HUB75-MatrixPanel-DMA-ICN2053#6] [Ideas]
Currently flipDMABuffer waits if bufferReady is false, if data is currently send to the display.

This waits with the drawing of the next frame, producing a syncronized drawing, making sure each frame is drawn, even it makes drawing slower.

In my use case I get external data and when drawing is too slow, it runs out of sync, producing serial buffer issues (new frames delivered by serial port). It would be better to skip frames, but to make sure display always shows the latest frame.

I solved that by adding a new function :
IRAM_ATTR bool MatrixPanel_DMA::flipDMABufferIfReady() 
{
  if (bufferReady)
  {  flipBuffer();
     return false;
  }
  else 
    return true;
}

Do you think that would be useful for your work as well? I could send a pull request, just want to make sure it fit's to your concept.

**Comment by LAutour** [comment]
I'll think about it, but it will be through a parameter in the standard SandFrame\SwapBuffer .
